store.go

```diff
@@ -2934,11 +2978,35 @@ func (s *store) Mounted(id string) (int, error) {
 }
 
 func (s *store) UnmountImage(id string, force bool) (bool, error) {
+	if err := s.startUsingGraphDriver(); err != nil {
+		return false, err
+	}
+	defer s.stopUsingGraphDriver()
+
+	if err := s.imageStore.startReading(); err != nil {
+		return false, err
+	}
+	defer s.imageStore.stopReading()
+
 	img, err := s.Image(id)
```

Comment:
`s.Image` does its own locking, this is unexpected.

Blocker: And the locking order now violates the locking hierarchy.

I’m fine with calling ordinary `s.Image()` here and only doing manual locking for layers afterwards; that’s much simpler, and no worse than what we did previously.